tests/src/podman-machine-onboarding.spec.ts

```diff
@@ -0,0 +1,143 @@
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let dashboardPage: DashboardPage;
+let resourcesPage: ResourcesPage;
+let settingsBar: SettingsBar;
+let navigationBar: NavigationBar;
+
+let notificationPodmanSetup: Locator;
+let onboardingStepBody: Locator;
+let podmanMachineConfiguration: Locator;
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('podman-extensions-e2e');
+
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  navigationBar = new NavigationBar(page);
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+}, 120000);
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+describe('Podman Machine initialization through Onboarding', async () => {
+  test('Check Podman set-up notification presence (should be visible if machine was not initialized)', async () => {
+    dashboardPage = await navigationBar.openDashboard();
+    await playExpect(dashboardPage.mainPage).toBeVisible();
+    await playExpect(dashboardPage.notificationsBox).toBeVisible();
+    notificationPodmanSetup = dashboardPage.notificationsBox
+      .getByRole('region')
+      .filter({ hasText: 'Podman needs to be set up' });
+    await playExpect(notificationPodmanSetup).toBeVisible();
+  });
+  test('Open Podman Machine Onboarding through Setup Notification', async () => {
+    await notificationPodmanSetup.getByTitle('Set up Podman').click();
+    await playExpect(page.getByRole('heading', { name: 'Podman Setup Header' })).toBeVisible();
+  });
+  test('Verify Podman Onboarding was started', async () => {
+    await playExpect(page.getByRole('heading', { name: 'Podman Setup Header' })).toBeVisible();
+  });
+  test('Go back to Dashboard', async () => {
+    dashboardPage = await navigationBar.openDashboard();
+    await playExpect(dashboardPage.mainPage).toBeVisible();
+  });
+  test('Re-Open Podman Machine Onboarding through Settings Resources page', async () => {
+    settingsBar = await navigationBar.openSettings();
+    await settingsBar.resourcesTab.click();
+    resourcesPage = new ResourcesPage(page);
+    const podmanResourcesSectionLocator: Locator = resourcesPage.getPage().getByLabel('podman', { exact: true });
+    await playExpect(podmanResourcesSectionLocator).toBeVisible();
+    await podmanResourcesSectionLocator.getByLabel('Setup Podman').click();
+    await playExpect(page.getByRole('heading', { name: 'Podman Setup Header' })).toBeVisible();
+    onboardingStepBody = page.locator('//*[@id="stepBody"]');
+    await playExpect(onboardingStepBody).toBeVisible();
+  });
+  test('Verify Podman Autostart is enabled and proceed to next page', async () => {
+    const podmanAutostartToggle: Locator = onboardingStepBody.getByLabel(
+      'Autostart Podman engine when launching Podman Desktop',
+    );
+    playExpect(await podmanAutostartToggle.isChecked()).toBeTruthy();
```

Comment:
PLaywright's expect should have particular assertion for `isChecked()`. No need to get the attribute and then test on truthy/falsy.